sv.c

```diff
@@ -4782,6 +4783,448 @@ Perl_sv_setsv_flags(pTHX_ SV *dsv, SV* ssv, const I32 flags)
+         * SVf_POK should not have been set on dsv either and so
+         * we assert that here. */
+        assert(!(SvFLAGS(dsv) & SVf_POK));
+    }
+
+    {
+        const char *vstr_pv;
+        STRLEN vstr_len;
+        if ((vstr_pv = SvVSTRING(ssv, vstr_len))) {  /* [ <<< 1% ] */
+            sv_magic(dsv, NULL, PERL_MAGIC_vstring, vstr_pv, vstr_len);
+            SvRMAGICAL_on(dsv);
+        }
+    }
+    if (SvTAINTED(ssv)) /* [ <<< 1% ] */
+        SvTAINT(dsv);
+    return dsv;
+}
+
+/*
+=for apidoc newSVsv_flags_NN
+
+This creates a new SV which contains the values of the original SV.
+
+It does the bulk of the work for C<newSVsv_flags> and C<sv_mortalcopy_flags>.
+Less common cases are passed to C<sv_setsv_flags>.
+
+This function accepts the same flags as C<sv_setsv_flags>, with the single
+addition of C<SVs_TEMP>, which toggles the treatment of a freed source SV:
+  * Not present: emulate C<newSVsv_flags> by emitting a warning and
+                 returning NULL.
+  * Present:     emulate C<sv_mortalcopy_flags> behaviour: C<croak()>.
+
+=cut
+*/
+
+    /* Note: If a PVIV/PVNV/PVMG is only IOK, NOK, ROK, it is _mostly_
+     * possible to create just a headless SV to store that value.
+     * Some parts of core (Perl_amagic_call in gv.c specifically) do
+     * assume - and possibly CPAN might - that SvTYPE(dsv) == SvTYPE(ssv)
+     * though, which is why the code below does not try that type
+     * simplification. Perhaps this might be worth revisiting in the future.
+     * -- April 2025. */
+/*
+ * [%] numbers are a rough percentage of calls to this function, as
+ * measured by a gcov build running the test harness. They are presented
+ * only for general information and will not be representative of all
+ * workloads or applications.
+*/
+
+SV *
+Perl_newSVsv_flags_NN(pTHX_ SV *const old, I32 flags)
+{
+    PERL_ARGS_ASSERT_NEWSVSV_FLAGS_NN;
+    SV *dsv;
+    new_SV(dsv);
+
+    /* new_SV includes default initialization of SvFLAGS and SvANY.
+     * However, the SVt_IV cases in the switch below are both very common
+     * and very simple. If we initialize for those cases, a decent compiler
+     * will hopefully elide new_SVs defaults so no extra work is done.
+     * For all other cases, SvANY and SvFLAGS will be overwritten anyway. */
+    SET_SVANY_FOR_BODYLESS_IV(dsv);
+    SvFLAGS(dsv) = SVt_IV|SVf_IOK|SVp_IOK;
+
+    const U32 sflags = SvFLAGS(old);
+
+    /* These are the hottest and simplest cases */
+    switch( sflags & (SVTYPEMASK|SVf_IOK|SVf_IVisUV|SVf_ROK|SVf_NOK|SVf_POK) ) {
+        case SVt_IV|SVf_IOK: /* [ 31% ] */
+            assert(SvANY(dsv));
+            assert(SvFLAGS(dsv) == (SVt_IV|SVf_IOK|SVp_IOK) );
+
+            assert(    &(old->sv_u.svu_iv)
+                == &(((XPVIV*) SvANY(old))->xiv_iv));
+            assert(    &(dsv->sv_u.svu_iv)
+                == &(((XPVIV*) SvANY(dsv))->xiv_iv));
+            dsv->sv_u.svu_iv = old->sv_u.svu_iv;
+            break;
```

Comment:
Why is there so much hate for the step-child of  the sv.c API, its name is `SVf_IVisUV`.  aren't SvIVX/SvUVX same length, 2s complements overlapping  slots with no risk of sign extension